samza-core/src/main/scala/org/apache/samza/metrics/MetricsRegistryMap.scala

```diff
@@ -51,7 +51,10 @@ class MetricsRegistryMap(val name: String) extends ReadableMetricsRegistry with
 
   def newGauge[T](group: String, gauge: Gauge[T]) = {
     debug("Adding new gauge %s %s %s." format (group, gauge.getName, gauge))
-    putAndGetGroup(group).putIfAbsent(gauge.getName, gauge)
+    if (putAndGetGroup(group).containsKey(gauge.getName)) {
+      info("Updating existing gauge %s %s %s" format (group, gauge.getName, gauge))
+    }
+    putAndGetGroup(group).put(gauge.getName, gauge)
```

Comment:
Should we do this for all other metric types -- timers, etc, and methods to create them. 
Last metric overrides -- Is this not the desired semantics?